lib/Mongo/MongoDB.php

```diff
@@ -405,7 +405,12 @@ public function command(array $data, $options = [], &$hash = null)
     {
         try {
             $cursor = new \MongoCommandCursor($this->connection, $this->name, $data);
-            $cursor->setReadPreference($this->getReadPreference());
+
+            if (!empty($data['new']) || !empty($data['upsert'])) {
```

Comment:
Note that there are many more [commands that write](https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#other-commands-that-write) than `findAndModify`, which it looks like this is attempting to detect.

Also, command detection would be better implemented by checking the first key in the command document (i.e. `$data` argument), as that must indicate the command name. That would allow most write commands to be detected, excluding `aggregate` and `mapReduce` where you would need to inspect additional options to determine if they are directing output to a collection.